src/tests/multihost/ad/test_dyndns.py

```diff
@@ -223,90 +250,370 @@ def test_0005_check_dyndns_iface_with_non_existing_interfaces(multihost, adjoin,
+        assert cmd_lookup_ptr.returncode == 0
+        
+    def test_0011_verify_use_after_free_in_dyndns_code_bz1132361(self, multihost, adjoin):
+        """
+        :title: IDM-SSSD-TC: ad_provider: dyndns: verify use after free in dyndns code bz1132361
+        bz1132361 is an OOM error that is recreated by
+        setting TALLOC_FREE_FILL
+        :id: 72d5e4ad-fdb9-4278-860f-57ff563a20d7
+        :steps:
+        1. Join client to AD domain
+        2. Add extra interface
+        3. Set TALLOC_FREE_FILL=253 in /etc/sysconfig/sssd
+        4. Start SSSD
+        :expectedresults:
+        1. Client joins domain
+        2. Extra interface is created
+        3. Configuration files updated
+        4. SSSD starts and does not segfault
+        """
+        adjoin(membersw='adcli')
+        client = sssdTools(multihost.client[0], multihost.ad[0])
+
+        multihost.client[0].run_command('echo "TALLOC_FREE_FILL=253" >> /etc/sysconfig/sssd')
+        client.clear_sssd_cache()
+        
+        ps_cmd = multihost.client[0].run_command('ps aux | grep sssd')
+        assert 'sssd_be' in ps_cmd.stdout_text
+        assert 'sssd_nss' in ps_cmd.stdout_text
+        assert 'sssd_pam' in ps_cmd.stdout_text
+        assert 'sssd_pac' in ps_cmd.stdout_text
+
+        multihost.client[0].run_command('sed -i "/TALLOC_FREE_FILL/d" /etc/sysconfig/sssd')
+
+    def test_0012_set_dyndns_update_ptr_when_dyndns_server_equals_ad_server(self, multihost, adjoin):
+        """
+        :title: IDM-SSSD-TC: ad_provider: dyndns: set dyndns update ptr when dyndns server equals ad server
+        Testing parameter ad_server = $AD_SERVER
+        :id: afa91b9b-4943-423a-a2e2-54f6c6204107
+        :steps:
+        1. Join client to AD domain
+        2. Configure SSSD to contain dyndns_server = AD_SERVER
+        3. Start SSSD
+        4. Check DNS records
+        :expectedresults:
+        1. Client joins domain
+        2. SSSD is configured
+        3. SSSD is started
+        4. Both A and PTR records are found
+        """
+        adjoin(membersw='adcli')
+        client = sssdTools(multihost.client[0], multihost.ad[0])
+        hostname_cmd = multihost.client[0].run_command('hostname', raiseonerr=False)
+        hostname = hostname_cmd.stdout_text.rstrip()
+        domain = multihost.ad[0].domainname
+        adserver = multihost.ad[0].hostname
+        ip = multihost.client[0].ip
+
+        domain_section = 'domain/{}'.format(domain)
+        sssd_params = {'dyndns_server': f'{adserver}'}
+        client.sssd_conf(domain_section, sssd_params)
+        client.clear_sssd_cache()
+
+        cmd_lookup = multihost.client[0].run_command(f"nslookup {hostname}", raiseonerr=False)
+        assert cmd_lookup.returncode == 0
+        cmd_lookup_ptr = multihost.client[0].run_command(f"nslookup {ip}", raiseonerr=False)
+        assert cmd_lookup_ptr.returncode == 0
 
     @classmethod
     def class_teardown(cls, multihost):
```

Comment:
There is no implementation of "removing zone files."

Reply:
Removing the comment, or should the function be removed?